worlds/hexcellsinfinite/__init__.py

```diff
@@ -0,0 +1,104 @@
+# If you want more things from the import, add a comma like the worlds.AutoWorld import below
+import logging
+
+from BaseClasses import MultiWorld, Item, Tutorial
+from worlds.AutoWorld import World, WebWorld
+from .Locations import get_location_names, get_total_locations
+from .Items import create_item, create_itempool, item_table, HEXCELLS_LEVEL_ITEMS 
+from .Options import HexcellsInfiniteOptions
+from .Regions import create_regions
+from .Rules import set_rules
+
+# This is where you setup the page on the site!
+# Typically is the name of your game with "WebWorld"
+# Whatever you named the folder you are holding all of this in
+class HexcellsInfiniteWebWorld(WebWorld):
+    # Theres a few different themes so have fun with it
+    theme = "partyTime"
+    
+    # You shouldnt have to change much here except the name at the bottom!
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up Hexcells Infinite for Archipelago. "
+        "This guide covers single-player, multiworld, and related software.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["ExpandedReality"]
+    )]
+
+# This class is the real meat and potatoes
+# Same as the first class, it's normally named whatever you named your folder with "World" at the end
+class HexcellsInfiniteWorld(World):
+    """
+    Hexcells Infinite is a deterministic minesweeper puzzle game. \nIf you like Minesweeper, but hate guessing, this is the game for you.
+    """
+    
+    # You want to put the full name of the game here. If you shortened the name for the folder and class names, dont do that here
+    game = "Hexcells Infinite"
+    # The item_table will be setup in your Items.py. This line gets all the items you put into item_table and puts it in a way that AP can understand it
+    item_name_to_id = {name: data.ap_code for name, data in item_table.items()}
+    # get_location_names() will come from your Locations.py
+    location_name_to_id = get_location_names()
+    # And these 2 are the name of your Options.py class. 
+    options_dataclass = HexcellsInfiniteOptions
+    options: HexcellsInfiniteOptions
+    # The name of the class above
+    web = HexcellsInfiniteWebWorld()
+
+    # This is where you'd do things just before the generation
+    # Super important for doing things like adjusting the item pool based on options and the like
+    # Can technically be skipped if you don't need to do anything, or if you handle it elsewhere like A Short Hike
+    # I highly recommend looking at other apworlds init files to see some examples
+    def generate_early(self):
+        # Push precollected is how you give your player items they need to start with
+        # This can be dependent based on your Options, like I do below 
+        # Don't worry about the starting inventory option, that's in all yamls. AP handles that one
+
+        if self.options.LevelUnlockType == Options.LevelUnlockType.option_individual :
```

Comment:
The `Options` here isn't imported directly, and relies on being implicitly imported in `Items.py`/`Rules.py`. Either there should be an import of it in this file, or you should import `LevelUnlockType` directly and remove it.
```suggestion
        if self.options.LevelUnlockType == LevelUnlockType.option_individual:
```
Also there's a fair amount of places like here that have extra whitespace.